java/flight/flight-core/src/test/java/org/apache/arrow/flight/client/TestCookieHandling.java

```diff
@@ -138,27 +164,75 @@ public void multipleCookiesWithSetCookie() {
     headersToSend.insert(SET_COOKIE_HEADER, "firstKey=firstVal");
     headersToSend.insert(SET_COOKIE_HEADER, "secondKey=secondVal");
     cookieMiddleware.onHeadersReceived(headersToSend);
-    Assert.assertEquals("firstKey=firstVal; secondKey=secondVal", cookieMiddleware.calculateCookieString());
+    Assert.assertEquals("firstKey=firstVal; secondKey=secondVal", cookieMiddleware.getValidCookiesAsString());
   }
 
   @Test
-  public void basicCookiesWithSetCookie2() {
-    CallHeaders headersToSend = new ErrorFlightMetadata();
-    headersToSend.insert(SET_COOKIE2_HEADER, "firstKey=firstVal");
-    cookieMiddleware.onHeadersReceived(headersToSend);
-    Assert.assertEquals("firstKey=firstVal", cookieMiddleware.calculateCookieString());
+  public void cookieStaysAfterMultipleRequestsEndToEnd() {
+    client.handshake();
+    Assert.assertEquals("k=v", cookieMiddleware.getValidCookiesAsString());
```

Comment:
These tests are wrong. You are just checking getValidCookiesAsString() on the same cookieMiddleware instead of new ones the server makes.

Reply:
Done, updated the test co check getValidCookiesAsString on newly created middlewares.